src/player.cpp

```diff
@@ -4160,14 +4160,14 @@ void player::pause()
     search_surroundings();
 }
 
-void player::shout( std::string msg )
+void player::shout( std::string msg, const std::string &verbal_shout_id )
 {
-    int base = 10;
+    int base = 85;
```

Comment:
> You started this PR when I enumerated the problems with limiting NPC "signals" on map to line of sight.

I've no idea what you mean by NPC "signals". I started this PR to address the concerns you raised in #16430, which I quoted a few comments above for your reference. Reread that PR to understand the context of this PR please.

> if this is supposed to be just an extra feature, it's OK. But it does not solve any real problem with NPCs.

It doesn't solve any existing problems with NPCs. It addresses the concerns you raised in #16430. Specifically, it provides a means to locate NPCs that are nearby that have wandered off a short distance.

> It does nothing to address that issue (case 2). Nothing at all. Not the slightest bit.

Reread my last comment. I didn't say this PR addresses case 2. #16430 addresses case 2. This PR addressed case 1 that you described above. FYI, here is your case 1, as you described it:

> those who come to you to talk to you/attack you

Coming to you to talk/attack is exactly the intended behavior.